src/docfx/lib/schema/JsonSchemaTransform.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Docs.Build
+{
+    internal static class JsonSchemaTransform
+    {
+        public static (List<Error> errors, JToken token) Transform(Document file, Context context, JsonSchema schema, JToken token, Action<Document> buildChild)
+        {
+            var errors = new List<Error>();
+
+            var transformedToken = JsonUtility.DeepClone(token);
```

Comment:
we could do an in place transform for perf reasons without this deep clone, if `JToken` is removed from output, it is clear from method signature that it is an in place transform.

Reply:
sounds good